File: outputs/2025-08-11_14-11-03-996988/TaskManager.py
# Unit test result: ✅ PASSED
from datetime import datetime

class Task:

    def __init__(self, description: str, priority: int):
        self.description = description
        self.priority = priority
        self.completed = False
        self.creation_date = datetime.now().strftime('%Y-%m-%d %H:%M')

    def get_description(self) -> str:
        return self.description

    def get_priority(self) -> int:
        return self.priority

    def is_completed(self) -> bool:
        return self.completed

    def set_completed(self, completed: bool) -> None:
        self.completed = completed

    def get_creation_date(self) -> str:
        return self.creation_date
    description: str
    completed: bool
    priority: int
    creationDate: str

    def __init__(self, description: str, priority: int) -> None:
        self.description = description
        self.completed = False
        self.priority = priority
        now = datetime.now()
        self.creationDate = now.strftime('%Y-%m-%d %H:%M')

    def get_description(self) -> str:
        return self.description

    def is_completed(self) -> bool:
        return self.completed

    def set_completed(self, completed: bool) -> None:
        self.completed = completed

    def get_priority(self) -> int:
        return self.priority

    def get_creation_date(self) -> str:
        return self.creationDate

class TaskManager:

    def __init__(self):
        self.tasks: list['Task'] = []

    def add_task(self, description: str, priority: int) -> None:
        new_task = Task(description, priority)
        self.tasks.append(new_task)
        print('Task added successfully!')

    def view_tasks(self) -> None:
        if not self.tasks:
            print('No tasks available.')
            return
        print('\n===== TASK LIST =====')
        for (i, task) in enumerate(self.tasks):
            status = '[COMPLETED]' if task.is_completed() else '[PENDING]'
            print(f'{i + 1}. {status} - {task.get_description()} (Priority: {task.get_priority()}) Created: {task.get_creation_date()}')
        print('=====================\n')

    def mark_task_as_complete(self, task_index: int) -> None:
        if 0 <= task_index < len(self.tasks):
            task = self.tasks[task_index]
            task.set_completed(True)
            print('Task marked as complete!')
        else:
            print('Invalid task number.')

    def run(self) -> None:
        running = True
        print('Welcome to Task Manager!')
        while running:
            print('\nPlease select an option:')
            print('1. Add new task')
            print('2. View all tasks')
            print('3. Mark task as complete')
            print('4. Exit')
            try:
                choice = int(input('Your choice: '))
            except ValueError:
                print('Invalid input. Please enter a number.')
                continue
            match choice:
                case 1:
                    description = input('Enter task description: ')
                    priority = int(input('Enter priority (1-5): '))
                    self.add_task(description, priority)
                case 2:
                    self.view_tasks()
                case 3:
                    self.view_tasks()
                    try:
                        task_num = int(input('Enter task number to mark as complete: '))
                        self.mark_task_as_complete(task_num - 1)
                    except ValueError:
                        print('Invalid task number input. Please enter a number.')
                    except IndexError:
                        print('Task number out of range.')
                case 4:
                    running = False
                    print('Thank you for using Task Manager!')
                case _:
                    print('Invalid option. Please try again.')

    @staticmethod
    def main(args: list[str]) -> None:
        manager = TaskManager()
        manager.run()
if __name__ == '__main__':
    TaskManager.main([])

File: uploads/2025-08-11_14-11-03-996988/TaskManager.java
import java.util.ArrayList;
import java.util.Scanner;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class TaskManager {
    private ArrayList<Task> tasks;
    private Scanner scanner;
    
    public TaskManager() {
        tasks = new ArrayList<>();
        scanner = new Scanner(System.in);
    }
    
    public void addTask(String description, int priority) {
        Task newTask = new Task(description, priority);
        tasks.add(newTask);
        System.out.println("Task added successfully!");
    }
    
    public void viewTasks() {
        if (tasks.isEmpty()) {
            System.out.println("No tasks available.");
            return;
        }
        
        System.out.println("\n===== TASK LIST =====");
        for (int i = 0; i < tasks.size(); i++) {
            Task task = tasks.get(i);
            String status = task.isCompleted() ? "[COMPLETED]" : "[PENDING]";
            System.out.printf("%d. %s - %s (Priority: %d) Created: %s\n", 
                i + 1, status, task.getDescription(), task.getPriority(), 
                task.getCreationDate());
        }
        System.out.println("=====================\n");
    }
    
    public void markTaskAsComplete(int taskIndex) {
        if (taskIndex >= 0 && taskIndex < tasks.size()) {
            Task task = tasks.get(taskIndex);
            task.setCompleted(true);
            System.out.println("Task marked as complete!");
        } else {
            System.out.println("Invalid task number.");
        }
    }
    
    public void run() {
        boolean running = true;
        
        System.out.println("Welcome to Task Manager!");
        
        while (running) {
            System.out.println("\nPlease select an option:");
            System.out.println("1. Add new task");
            System.out.println("2. View all tasks");
            System.out.println("3. Mark task as complete");
            System.out.println("4. Exit");
            System.out.print("Your choice: ");
            
            int choice = scanner.nextInt();
            scanner.nextLine(); // Consume newline
            
            switch (choice) {
                case 1:
                    System.out.print("Enter task description: ");
                    String description = scanner.nextLine();
                    System.out.print("Enter priority (1-5): ");
                    int priority = scanner.nextInt();
                    scanner.nextLine(); // Consume newline
                    addTask(description, priority);
                    break;
                case 2:
                    viewTasks();
                    break;
                case 3:
                    viewTasks();
                    System.out.print("Enter task number to mark as complete: ");
                    int taskNum = scanner.nextInt();
                    scanner.nextLine(); // Consume newline
                    markTaskAsComplete(taskNum - 1);
                    break;
                case 4:
                    running = false;
                    System.out.println("Thank you for using Task Manager!");
                    break;
                default:
                    System.out.println("Invalid option. Please try again.");
            }
        }
        
        scanner.close();
    }
    
    public static void main(String[] args) {
        TaskManager manager = new TaskManager();
        manager.run();
    }
}

class Task {
    private String description;
    private boolean completed;
    private int priority;
    private String creationDate;
    
    public Task(String description, int priority) {
        this.description = description;
        this.completed = false;
        this.priority = priority;
        
        LocalDateTime now = LocalDateTime.now();
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
        this.creationDate = now.format(formatter);
    }
    
    public String getDescription() {
        return description;
    }
    
    public boolean isCompleted() {
        return completed;
    }
    
    public void setCompleted(boolean completed) {
        this.completed = completed;
    }
    
    public int getPriority() {
        return this.priority;
    }
    
    public String getCreationDate() {
        return creationDate;
    }
}